contrib/packaging/docker/iptables-wrapper

```diff
@@ -0,0 +1,45 @@
+#!/bin/sh
+
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+# Detect whether the base system is using iptables-legacy or
+# iptables-nft. This assumes that some non-containerized process (eg
+# kubelet) has already created some iptables rules.
+
+# Bugs in iptables-nft 1.8.3 may cause it to get stuck in a loop in
+# some circumstances, so we have to run the nft check in a timeout. To
+# avoid hitting that timeout, we only bother to even check nft if
+# legacy iptables was empty / mostly empty.
+
+num_legacy_lines=$( (iptables-legacy-save || true; ip6tables-legacy-save || true) 2>/dev/null | grep '^-' | wc -l)
+if [ "${num_legacy_lines}" -ge 10 ]; then
+    mode=legacy
+else
+    num_nft_lines=$( (timeout 5 sh -c "iptables-nft-save; ip6tables-nft-save" || true) 2>/dev/null | grep '^-' | wc -l)
+    if [ "${num_legacy_lines}" -ge "${num_nft_lines}" ]; then
+	mode=legacy
+    else
+	mode=nft
+    fi
+fi
+
+update-alternatives --set iptables "/usr/sbin/iptables-${mode}" > /dev/null
+update-alternatives --set ip6tables "/usr/sbin/ip6tables-${mode}" > /dev/null
+
```

Comment:
How do these lines work for `ip6?tables-save` and `ip6?tables-restore`, which are also mapped to this script?

Reply:
Not sure about that, but from what I have tested on nft-enabled node `iptables-save` properly returns list of rules for the appropriate mode. I assume iptables-restore will work same way.

Reply:
To me it seems that while `iptables` and `ip6tables` are un-wrapped by the wrapper once the mode is detected, `iptables-save`, `ip6tables-save`, `iptables-restore`, and `ip6tables-restore` will remain wrapped, execute the detection, and `update-alternatives` on both `iptables` and `ip6tables` on each invocation. It would seem simple to "unwrap" all 6 instead of just the 2 above.

Reply:
Because of how the master-slave mechanism works in `alternatives`, we dont need to update the other two cmds (`*-save` and `*-restore`) and, most importantly, those get un-wrapped together with `ip(6)tables` cmds.

For more clarity, this is set of `update-alternatives --display iptables` outputs:
1. default (right after installing iptables in system):
```
iptables - auto mode
  link best version is /usr/sbin/iptables-legacy
  link currently points to /usr/sbin/iptables-legacy
  link iptables is /usr/sbin/iptables
  slave iptables-restore is /usr/sbin/iptables-restore
  slave iptables-save is /usr/sbin/iptables-save
/usr/sbin/iptables-legacy - priority 20
  slave iptables-restore: /usr/sbin/iptables-legacy-restore
  slave iptables-save: /usr/sbin/iptables-legacy-save
/usr/sbin/iptables-nft - priority 10
  slave iptables-restore: /usr/sbin/iptables-nft-restore
  slave iptables-save: /usr/sbin/iptables-nft-save
```

2. Initial state of cilium runtime container (`/usr/sbin/iptables-wrapper` not run yet):
```
iptables - auto mode
  link best version is /usr/sbin/iptables-wrapper
  link currently points to /usr/sbin/iptables-wrapper
  link iptables is /usr/sbin/iptables
  slave iptables-restore is /usr/sbin/iptables-restore
  slave iptables-save is /usr/sbin/iptables-save
/usr/sbin/iptables-legacy - priority 20
  slave iptables-restore: /usr/sbin/iptables-legacy-restore
  slave iptables-save: /usr/sbin/iptables-legacy-save
/usr/sbin/iptables-nft - priority 10
  slave iptables-restore: /usr/sbin/iptables-nft-restore
  slave iptables-save: /usr/sbin/iptables-nft-save
/usr/sbin/iptables-wrapper - priority 100
  slave iptables-restore: /usr/sbin/iptables-wrapper
  slave iptables-save: /usr/sbin/iptables-wrapper
```

3. post (`/usr/sbin/iptables-wrapper` run at least once):
```
iptables - manual mode
  link best version is /usr/sbin/iptables-wrapper
  link currently points to /usr/sbin/iptables-nft
  link iptables is /usr/sbin/iptables
  slave iptables-restore is /usr/sbin/iptables-restore
  slave iptables-save is /usr/sbin/iptables-save
/usr/sbin/iptables-legacy - priority 20
  slave iptables-restore: /usr/sbin/iptables-legacy-restore
  slave iptables-save: /usr/sbin/iptables-legacy-save
/usr/sbin/iptables-nft - priority 10
  slave iptables-restore: /usr/sbin/iptables-nft-restore
  slave iptables-save: /usr/sbin/iptables-nft-save
/usr/sbin/iptables-wrapper - priority 100
  slave iptables-restore: /usr/sbin/iptables-wrapper
  slave iptables-save: /usr/sbin/iptables-wrapper
```